vehicle/OVMS.V3/components/vehicle_mgev/src/mg_gwm_auth.cpp

```diff
@@ -139,43 +141,51 @@ void OvmsVehicleMgEv::GwmAuthentication(canbus* currentBus, uint8_t frameType, u
             // Seed1 response
             uint32_t seed = (data[1] << 24) | (data[2] << 16) | (data[3] << 8) | data[4];
             uint32_t key = pass1(seed);
-            nextFrame.data.u8[1] = 0x42u;
-            nextFrame.data.u8[2] = key >> 24u;
-            nextFrame.data.u8[3] = key >> 16u;
-            nextFrame.data.u8[4] = key >> 8u;
-            nextFrame.data.u8[5] = key;
-            nextFrame.FIR.B.DLC = 6u;
+            std::stringstream sSeed, sKey;
+            sSeed << std::hex << seed;
+            sKey << std::hex << key;
+            ESP_LOGV(TAG, "GWM auth: seed1 received %s. Replying with key1 %s", sSeed.str().c_str(), sKey.str().c_str());
```

Comment:
Just use %x rather than taking stringstream.

Reply:
Ok, not so familiar with c++ strings, will check out %x.